### ISSUE TITLE
External SMB storage which is not 7x24 available as my ownCloud server

### ISSUE THREAD

**Issue by peossource** [owncloud/core#18942]
Hello ownCloud developers,

after starting a thread at https://forum.owncloud.org/viewtopic.php?f=31&t=30320 I was told to open an issue here because it's not sure if this behavior is normal. On the other hand this can be an improvement or feature request.

When an already configured and working SMB share is not available during normal ownCloud operation then the server gets a lot of active apache2 zombie processes for a longer time. Every access to the share I assume creates a new process e.g. an admin page trying to check connectivity, user access and I also assume the background cron too. This is to much for my Raspberry Pi (single core, 512 MB RAM). These zombies are allocating more and more RAM, swapping begins...
It doesn't matter if the share is already unavailable at boot time or gets offline during normal operation.

Is ownCloud not able /not designed for to handle non 7x24 SMB shares what I think is a normal use case in private environments ? My NAS is not always up and I don't want to have dependencies.

My expected behavior is that ownCloud detects an offline share. Only tries a shorter time to access it (e.g. 20s) and when it detects an offline share then the next time consuming real try to connect should start only after a wait time (e.g. 5 mins) independent to still existing access tries.
An offline share should be reported e.g. showing a different folder icon with an red cross.
And clients should know that this part of file space is temporarily down and not deleted.

Best regards,
Christian
ownCloud 8.1 on Raspbian/Debian 7 (apache 2.2, php 5.4), mysql

**Comment by Xenopathic** [comment]
I believe you are running into an upstream smbclient bug, which doesn't properly close connections. cc @icewind1991

**Comment by peossource** [comment]
Ok. What should be the normal behavior when a SMB share is down? Sorry I'm new to ownCloud and I've started with 8.1.1.3 stable. Are there any differences in behavior between the current code and my expectations from the first post above? If yes they could be a feature request or improvement.

Please let me know if I can do further support on this bug!

Btw. ownClud is a root cause for my comeback to break in into the deeps of apache, php, mysql after >10 years of absence.

Best regards,
Christian

**Comment by peossource** [comment]
sorry, wrong button. This should left open

**Comment by Xenopathic** [comment]
Hmm, actually, this is a different problem. It seems that when the server isn't available connections are opened but not closed again cc @icewind1991

**Comment by Xenopathic** [comment]
@peossource Are you using libsmbclient-php?

**Comment by peossource** [comment]
No.
And I can't find it with apt-get on debian 7. There is only smbclient and libsmbclient installed (no php in the package name).
In the output of phpinfo() (apache2 and cli) there is nothing about smb.

**Comment by nidarian** [comment]
Hello,
This felt like a good place to mention that when my external SMB is offline, my owncloud 8.1 webdav for windows/android clients cannot connect. When I power on the smb server and the owncloud server can connect to it, then webdav starts working again.

**Comment by PVince81** [comment]
Closing in favor of https://github.com/owncloud/core/issues/22826

In general we need the SMB code in OC to report that the storage is not available so that the sync client and others can skip it.
